hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/ByteArrayObjectDataInput.java

```diff
@@ -385,6 +387,9 @@ public short readShort(int position, ByteOrder byteOrder) throws IOException {
     @Override
     public byte[] readByteArray() throws IOException {
         int len = readInt();
+        if (len == NULL_ARRAY) {
```

Comment:
I would rename NULL_ARRAY to NULL_ARRAY_LENGTH or NULL_LENGTH. 

Because NULL_ARRAY makes you believe that the field is of type byte[], but actually it is a length indicator. So it puts the reader on the wrong foot.